### PR TITLE
memesuite: update version to 5.4.1

### PR DESCRIPTION
**Pull request by tlawbailey**
New version (5.4.1) of the MEME Suite.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7
Xcode 11.3

macOS 11.6
Xcode 10.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by tlawbailey** [comment]
### MEME Suite Version 5.4.1
### New Features and Enhancements
        **XSTREME:** Combined motif discovery and enrichment analysis. Combines the results of ab initio discovery using MEME and STREME with motif enrichment analysis using SEA. Groups similar ab initio motifs with each other and with known motifs. XSTREME is similar to MEME-CHIP, but is more appropriate when the motifs may not tend to be enriched in the centers of the input sequences, and works well with protein sequences and nucleotide sequences.
        **SEA:** Simple Enrichment Analysis of motifs. Reports known motifs that are enriched in your sequences relative to your control sequences or to shuffled versions of your sequences. Replaces and enhances ENR.
        **STREME:**
            Added options --thresh and --evalue to allow STREME to stop looking for motifs based on their E-values rather than their p-values. (The --thresh option replaces the --pvt option, which is still available for backward compatability.)
            Added motif E-value column in HTML output and change MEME format motif header to E= in streme.txt.
            Added "Positional Distribution" plots to HTML output. These plots show how the sites of each motif are distributed within the input sequences. Alignment of sequences for the plots is controlled by the new --align option.
            Added output of a file (sequences.tsv) containing the names of each input sequence that contains a site of a discovered motif.
            Added "Matches per Sequence" plots to HTML output. These plots show the distribution of the percentage of sequences with different numbers of matches to each significant motif.
            The Sites column in the HTML now shows both the number of percentage of positive sequences with a site as well as the number of sequences.
            The meme.txt file of motifs created by STREME now shows the Score (Training Set p-value) rather than the Test Set p-value when there was no Test Set due to the input containing too few sequences. This is indicated in the motif "letter-probability matrix" line, where "P=" is now replaced by "S=".
        **MEME:** Will now search for up to 1000 motifs if -evt is given and you don't specify -nmotifs. Previously, it would only search for (up to) 1 motif.
        **AME:**
            Added the motif alternate ID to the sequence.tsv output. The new column in the TSV file is named "motif_ALT_ID".
            Added option --text to just send TSV output to standard output. 
        **fasta-holdout-set:** New utility for splitting primary (positive) and control (negative) sequence sets into a main and holdout sets. Input is one or two FASTA sequence files. Output is two or four FASTA sequence files. If no control sequences are given, they are created by shuffling the primary sequences. By default, each FASTA file (primary and control) is split into two FASTA files containing 90% ("main file") and 10% (holdout file) of the input sequences, respectively. If either of the holdout files (primary or control) would contain fewer than 10 sequences, no holdout files are created and all sequences are put in the main files. This utility was create for use by XSTREME.

### Bug Fixes
        **CentriMo:** fixed bug with "Download EPS (for publication)" button not working on some browsers.
        **GOMO:** fixed links from GO terms to their descriptions at "amigo".
        **MAST:** fixed bug in usage message.
        **MEME:**
            Fixed core dump caused with the "anr" model by very long sequences.
            The -searchsize option now also limits the length of the longest sequence allowed, unless you specify "-searchsize 0". 
        **STREME:**
            Fixed failure to honor "T=U" in custom RNA alphabets.
            Fixed error when maximum motif width is larger than average sequence length.
            Fixed printing of motifs in MEME format when there were no test positives or negatives but there was a hold-out set. 
        **fasta-get-markov:** was not deleting its temporary files /tmp/bfile.*.

File: science/memesuite/Portfile
Comment (pmetzger) [inline science/memesuite/Portfile:6]: `revision` should be kept (and set to 0 and properly indented).

Comment (pmetzger) [reply science/memesuite/Portfile:6]: (Also, we're missing the git handles of two of the maintainers.)

**Review by pmetzger — CHANGES_REQUESTED** [review]
Thanks for the update! A couple of small notes.

**Comment by pmetzger** [comment]
You indicated in the checklist that you followed our commit message guidelines, but the commit message doesn't (in fact) follow the guidelines. We will need that fixed as well before a merge.

**Comment by mascguy** [comment]
> You indicated in the checklist that you followed our commit message guidelines, but the commit message doesn't (in fact) follow the guidelines. We will need that fixed as well before a merge.

Agreed. But we'll certainly help you get these issues resolved!

File: science/memesuite/Portfile
Comment (mascguy) [reply science/memesuite/Portfile:6]: Fixed

**Review by mascguy — APPROVED** [review]

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
Updated the commit message, per our guidelines. And also took care of some minor cleanup.

Also tested locally, and all is well. Thank you so much for your contribution!

**Comment by mascguy** [comment]
Fellow Reviewers: It wasn't my intention to merge this prior to the 72-hour maintainer timeout. Instead, I was so focused on assisting @tlawbailey with getting this PR in ship-shape condition, that I was a bit hasty with the merge. So I sincerely apologize for that!

If there are any concerns with the changes, please let me know ASAP.

**Pull request merged**
